tests/sast/test_parser.py

```diff
@@ -0,0 +1,384 @@
+            'severity': 'LOW'}, 'scope': {'languages': ['python']},
+        'definition': [
+            {'cond_type': 'pattern', 'operator': 'equals', 'value': 'key = $KEY'},
+            {'cond_type': 'variable', 'variable': '$KEY', 'operator': 'regex_match', 'value': '^.*(RSA)/.*'},
+        ]
+    }
+    parser = SastCheckParser()
+    parsed_check = parser.parse_raw_check_to_semgrep(raw_check)
+    print (parsed_check)
+    assert parsed_check == {
+        'id': 'CKV_SAST_1', 'message': 'some guidelines', 'severity': 'INFO', 'languages': ['python'], 'metadata': {
+            'name': 'check name'},
+        'patterns': [
+            {'pattern': 'key = $KEY'},
+            {'metavariable-regex': {
+                'metavariable': '$KEY',
+                'regex': '^.*(RSA)/.*'
+                }
+            },
+        ]
+    }
+
+
+def test_metavariable_less_than_comparison_parsing():
+    raw_check = {
+        'metadata': {'id': 'CKV_SAST_1', 'name': 'check name', 'guidelines': 'some guidelines', 'category': 'sast',
+            'severity': 'LOW'}, 'scope': {'languages': ['python']},
+        'definition': [
+            {'cond_type': 'pattern', 'operator': 'equals', 'value': 'key = $KEY'},
+            {'cond_type': 'variable', 'variable': '$KEY', 'operator': 'less_than', 'value': '1024'},
+        ]
+    }
+    parser = SastCheckParser()
+    parsed_check = parser.parse_raw_check_to_semgrep(raw_check)
+    print (parsed_check)
+    assert parsed_check == {
+        'id': 'CKV_SAST_1', 'message': 'some guidelines', 'severity': 'INFO', 'languages': ['python'], 'metadata': {
+            'name': 'check name'},
+        'patterns': [
+            {'pattern': 'key = $KEY'},
+            {'metavariable-comparison': {
+                'metavariable': '$KEY',
+                'comparison': '$KEY < 1024'
+                }
+            },
+        ]
+    }
+
+
+def test_metavariable_greater_than_comparison_parsing():
+    raw_check = {
+        'metadata': {'id': 'CKV_SAST_1', 'name': 'check name', 'guidelines': 'some guidelines', 'category': 'sast',
+            'severity': 'LOW'}, 'scope': {'languages': ['python']},
+        'definition': [
+            {'cond_type': 'pattern', 'operator': 'equals', 'value': 'key = $KEY'},
+            {'cond_type': 'variable', 'variable': '$KEY', 'operator': 'greater_than', 'value': '1024'},
+        ]
+    }
+    parser = SastCheckParser()
+    parsed_check = parser.parse_raw_check_to_semgrep(raw_check)
+    print (parsed_check)
+    assert parsed_check == {
+        'id': 'CKV_SAST_1', 'message': 'some guidelines', 'severity': 'INFO', 'languages': ['python'], 'metadata': {
+            'name': 'check name'},
+        'patterns': [
+            {'pattern': 'key = $KEY'},
+            {'metavariable-comparison': {
+                'metavariable': '$KEY',
+                'comparison': '$KEY > 1024'
+                }
+            },
+        ]
+    }
+
+
+def test_complex_policy_parsing():
```

Comment:
this entire file looks like one big "pytest.mark.parametrize" to me.
however I do like the different test names as it is much more clear like that, but I do think you can name test cases in pytest parametrize - there is a package called "Parametrization" or something like that which supports it I believe.

Again no need to do it now as it is already in progress, but it's good for general knowledfe

Reply:
I just think writing every test separately like this is the most readable way to do it, but if you have another example that looks readable and has other advantages it'd be nice to change a lot of our tests to it